kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -560,29 +300,39 @@ public open class ApplicationCommandRegistry : KoinComponent {
         val response = if (guild == null) {
             // We're registering global commands here, if the guild is null
 
-            kord.createGlobalUserCommand(name) {
+            kord.createGlobalMessageCommand(name) {
                 logger.trace { "Adding/updating global ${command.type.name} command: $name" }
 
                 this.register(locale, command)
             }
         } else {
             // We're registering guild-specific commands here, if the guild is available
 
-            guild.createUserCommand(name) {
+            guild.createMessageCommand(name) {
                 logger.trace { "Adding/updating guild-specific ${command.type.name} command: $name" }
 
                 this.register(locale, command)
             }
         }
 
+        injectPermissions(guild, command, response.id) ?: return null
+
+        return response.id
+    }
+
+    // endregion
+
+    // region: Permissions
+
+    protected suspend fun <T : ApplicationCommand<*>> injectPermissions(
```

Comment:
Nope, this needs to be public for third-party subtypes.

Reply:
It is

Reply:
Yup yup, visibility modifier confusion as mentioned on Discord